driver-core/src/main/java/com/datastax/driver/core/CBUtil.java

```diff
@@ -242,12 +242,16 @@ public static int sizeOfStringToStringListMap(Map<String, List<String>> m) {
         return size;
     }
 
-    public static ByteBuffer readValue(ChannelBuffer cb) {
+    public static ByteBuffer readValue(ByteBuf cb) {
         int length = cb.readInt();
-        return length < 0 ? null : cb.readSlice(length).toByteBuffer();
+        if (length < 0)
+            return null;
+        ByteBuf slice = cb.readSlice(length);
+
+        return ByteBuffer.wrap(readRawBytes(slice));
```

Comment:
This makes a copy though, whereas slice.nioBuffer() would share the contents.
We use this method when decoding result frames. Among other things, it extracts raw column values (that will ultimately be exposed to the client) from the frame body, which is a pooled buffer. That buffer is released just after decoding, so we want to keep the copying.